src/brod_producer_sup.erl

```diff
@@ -0,0 +1,59 @@
+%%%
+%%%   Copyright (c) 2015 Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+
+%%%=============================================================================
+%%% @doc
+%%% @copyright 2015 Klarna AB
+%%% @end
+%%%=============================================================================
+-module(brod_producer_sup).
+
+-behaviour(supervisor).
+
+%% API
+-export([ start_link/0
+        , start_worker/2
+        ]).
+
+%% Supervisor callbacks
+-export([init/1]).
+
+%%%_* API functions ============================================================
+start_link() ->
+  supervisor:start_link(?MODULE, []).
+
+start_worker(SupPid, Args) ->
+  supervisor:start_child(SupPid, Args).
+
+%%%_* Supervisor callbacks =====================================================
+init([]) ->
+  Spec = worker(brod_producer_worker, 5000, permanent),
```

Comment:
This is definitely not OK. There's no `start_link/0` in `brod_producer_worker`. Most probably this supervisor should only have dynamically added children.